source/common/filter/tcp_proxy.cc

```diff
@@ -217,29 +230,38 @@ Network::FilterStatus TcpProxy::initializeUpstreamConnection() {
       config_->stats().downstream_cx_no_route_.inc();
     }
     request_info_.setResponseFlag(AccessLog::ResponseFlag::NoRouteFound);
-    onInitFailure();
+    onInitFailure(Http::Code::ServiceUnavailable);
     return Network::FilterStatus::StopIteration;
   }
 
   Upstream::ClusterInfoConstSharedPtr cluster = thread_local_cluster->info();
   if (!cluster->resourceManager(Upstream::ResourcePriority::Default).connections().canCreate()) {
     request_info_.setResponseFlag(AccessLog::ResponseFlag::UpstreamOverflow);
     cluster->stats().upstream_cx_overflow_.inc();
-    onInitFailure();
+    onInitFailure(Http::Code::ServiceUnavailable);
     return Network::FilterStatus::StopIteration;
   }
+
+  const uint32_t max_connect_attempts = (config_ != nullptr) ? config_->maxConnectAttempts() : 1;
```

Comment:
not actionable for this review, but I agree this config null check is getting ugly and we should get rid of it. I remember thinking about it when this originally happened in the websocket review and letting it slide. oh well. Can you potentially add a TODO somewhere just to remind us that we need to do this?

Reply:
Yes, I'm adding a TODO at the top of the class.